tests/test_handler.py

```diff
@@ -0,0 +1,445 @@
+import importlib.machinery
+import importlib.util
+import logging
+import os
+import shutil
+import subprocess
+import textwrap
+import pytest
+from aws_cdk.cx_api import CloudAssembly
+from botocore.credentials import Credentials
+from boto3.session import Session
+from sceptre import exceptions
+from template_handler.cdk import CDK
+from unittest.mock import Mock, MagicMock
+from sceptre.connection_manager import ConnectionManager
+
+name = 'dev/cdk'
+sceptre_user_data = {'object_name': 'object-key.txt'}
+stack_group_config = {'project_code': 'sceptre'}
+arguments = {
+    'path': 'CDK/s3.py'
+}
+aws_profile = 'profile1'
+aws_region = 'eu-west-1'
+aws_iam_role = 'role1'
+
+aws_access_key = '123'
+aws_secret_key = '456'
+aws_token = '789'
+
+connection_manager = MagicMock(
+    spec=ConnectionManager
+)
+
+
+def test_schema():
+    """
+    Test schema function
+    """
+    # Arrange
+    cdk_handler = CDK(
+        name,
+        arguments=arguments,
+        sceptre_user_data=sceptre_user_data,
+        connection_manager=connection_manager,
+        stack_group_config=stack_group_config)
+
+    # Act
+    result = cdk_handler.schema()
+
+    # Assert
+    assert result['type'] == 'object'
+    assert result['properties'] is not None
+    assert result['required'] is not None
+
+
+def test_get_envs():
+    """
+    Test _get_envs function
+    """
+    # Arrange
+    credentials = Credentials(
+        access_key=aws_access_key,
+        secret_key=aws_secret_key
+    )
+
+    connection_manager = MagicMock(
+        spec=ConnectionManager,
+        profile=aws_profile,
+        region=aws_region,
+        iam_role=aws_iam_role
+    )
+
+    connection_manager._get_session = MagicMock(
+        spec_set=Session
+    )
+    connection_manager._get_session.return_value.get_credentials.return_value = credentials
```

Comment:
Pro tip: You can do this, which I find a bit cleaner

```suggestion
    connection_manager = MagicMock(**{
        'spec': ConnectionManager,
        'profile': aws_profile,
        'region': aws_region,
        'iam_role': aws_iam_role,
        '_get_session.return_value.spec_set': Session,
        '_get_session.return_value.get_credentials.return_value': credentials
    })
```